src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -96,10 +96,18 @@
   private Level tracingLevel = Level.INFO;
   // client to log session metrics to telemetry in GS
   private Telemetry telemetryClient;
-  // name of temporary stage to upload array binds to; null if none has been created yet
-  private String arrayBindStage = null;
   private SnowflakeConnectString sfConnStr;
 
+  // This constructor is used only by tests with no real connection.
+  // For real connections, the other constructor is always used.
+  public SFSession() {
```

Comment:
If it is test only, do we have notations like @VisibleForTesting

Reply:
Yes , will be better to make this protected instead of public.

Reply:
I think you mean package-private (done)? Annotation added.

Reply:
Actually, even tests from different packages try to use it.